### PR TITLE
BUGZ-967: Fix text visibility toggle

### PR DESCRIPTION
**Pull request by SamGondelman**
https://highfidelity.atlassian.net/browse/BUGZ-967

### CODE REVIEW

File: libraries/entities-renderer/src/RenderableEntityItem.cpp [hunk 1]
```diff
@@ -173,16 +173,21 @@ render::hifi::Layer EntityRenderer::getHifiRenderLayer() const {
...
     if (_canCastShadow) {
-        return ItemKey::Builder::opaqueShape().withTypeMeta().withTagBits(getTagMask()).withShadowCaster().withLayer(getHifiRenderLayer());
-    } else {
-        return ItemKey::Builder::opaqueShape().withTypeMeta().withTagBits(getTagMask()).withLayer(getHifiRenderLayer());
+        builder.withShadowCaster();
+    }
+
+    if (!_visible) {
+        builder.withInvisible();
     }
```
Comment (SamGondelman) [inline libraries/entities-renderer/src/RenderableEntityItem.cpp]: This new code is equivalent and repeats less code, and adds this missing visibility check (the actual fix)

File: libraries/entities-renderer/src/RenderableTextEntityItem.cpp [hunk 2]
```diff
@@ -282,20 +282,7 @@ ItemKey entities::TextPayload::getKey() const {
         auto renderable = entityTreeRenderer->renderableForEntityId(_entityID);
         if (renderable) {
             auto textRenderable = std::static_pointer_cast<TextEntityRenderer>(renderable);
-            ItemKey::Builder key;
-            // Similar to EntityRenderer::getKey()
```
Comment (SamGondelman) [inline libraries/entities-renderer/src/RenderableTextEntityItem.cpp]: let's just reuse it then!

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20301/HighFidelity-Beta-PR15905-9a866a1.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20301/HighFidelity-Beta-Interface-PR15905-9a866a1.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20301/HighFidelity-Beta-PR15905-9a866a1.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20301/HighFidelity-Beta-Interface-PR15905-9a866a1.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20301/HighFidelity-Beta-PR15905-9a866a1.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20301/HighFidelity-Quest-Beta-PR15905-9a866a1.apk)

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20311/HighFidelity-Beta-PR15905-a64d307.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20311/HighFidelity-Quest-Beta-PR15905-a64d307.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20311/HighFidelity-Beta-PR15905-a64d307.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20311/HighFidelity-Beta-Interface-PR15905-a64d307.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20311/HighFidelity-Beta-PR15905-a64d307.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20311/HighFidelity-Beta-Interface-PR15905-a64d307.exe

**Pull request merged**